File: team/verishot-extension/src/Extension.fs
module VerishotExtension.Extension

open VerishotExtension.Util
open VerishotExtension.Verishot

open Fable.Import
open Fable.Import.JS
open Fable.Import.Browser
open Fable.Import.vscode
open Fable.Core.JsInterop


let activate (context: vscode.ExtensionContext) =  
    printfn "Verishot Extension acivated"
    // vscode.commands.registerCommand("extension.helloWorld", fun _ -> 
    //     vscode.window.showInformationMessage "Hello!" |> unbox )
    // |> context.subscriptions.Add
    
    // Lint, Simulate, Visualise
    vscode.commands.registerCommand("extension.lint", fun _ ->
        execVerishot Lint)
    |> context.subscriptions.Add

    vscode.commands.registerCommand("extension.simulate", fun _ ->
        execVerishot Simulate)
    |> context.subscriptions.Add

    vscode.commands.registerCommand("extension.visualise", fun _ ->
        execVerishot Visualise)
    |> context.subscriptions.Add

File: team/Verishot-Extension/src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
import { execVerishot } from './verishot';
import { VerishotMode, verishotExists } from './utility';
import { newModuleHandler, newProjectHandler, deleteModuleHandler } from './project';
import { subscribeIntellisense } from './intellisense';

export const activate = (context: vscode.ExtensionContext) => {
	/// need to make sure `verishot` exists in PATH variable, otherwise throw an error and exit
	if (!verishotExists()) {
		vscode.window.showErrorMessage(`\`verishot\` is not found in your system. Please make sure it is installed and in your PATH variables.`);
		return;
	}

	// --lint, --simulate, --visualise
	const lint = vscode.commands.registerCommand('extension.lint', () => {
		execVerishot(VerishotMode.lint);
	});
	context.subscriptions.push(lint);

	const simulate = vscode.commands.registerCommand('extension.simulate', () => {
		execVerishot(VerishotMode.simulate);
	});
	context.subscriptions.push(simulate);

	const visualise = vscode.commands.registerCommand('extension.visualise', () => {
		execVerishot(VerishotMode.visualise);
	});
	context.subscriptions.push(visualise);





	// INTELLISENSE
	const diagnosticCollection = vscode.languages.createDiagnosticCollection();
	subscribeIntellisense(context, diagnosticCollection);




	// PROJECT
	const newproject = vscode.commands.registerCommand('extension.newproject', () => {
		newProjectHandler();
	});
	context.subscriptions.push(newproject);

	const newmodule = vscode.commands.registerCommand('extension.newmodule', () => {
		newModuleHandler();
	});
	context.subscriptions.push(newmodule);

	const deletemodule = vscode.commands.registerCommand('extension.deletemodule', () => {
		deleteModuleHandler();
	});
	context.subscriptions.push(deletemodule);

};

// this method is called when your extension is deactivated
export function deactivate() { }
